examples/image-to-text/run_pipeline.py

```diff
@@ -316,17 +316,35 @@ def main():
     if args.use_kv_cache:
         generate_kwargs["use_cache"] = args.use_kv_cache
 
+    if model_type == "chatglm":
+        generate_kwargs["reuse_cache"] = True
+
     if args.quant_config:
         generator.model = setup_quantization(generator.model, args)
         htcore.hpu_initialize(generator.model)
 
     # delete once pipeline integrate AutoProcessor as preprocess engine
-    if model_type in ["idefics2", "mllama", "paligemma"]:
+    if model_type in ["idefics2", "mllama", "paligemma", "chatglm"]:
         from transformers.image_utils import load_image
 
         def preprocess(self, image, prompt=None, timeout=None):
             image = load_image(image, timeout=timeout)
-            model_inputs = processor(images=image, text=prompt, return_tensors=self.framework)
+            if model_type == "chatglm":
+                query = "What is shown in this image?"
+                prompt = [{"role": "user", "image": image, "content": query}]
```

Comment:
```suggestion
                if prompt is None:
                    query = "What is shown in this image?"
                    prompt = [{"role": "user", "image": image, "content": query}]
```

If the user supplies a prompt

Reply:
GLM-4v has a special format for input prompt, I changed it to:
<pre>
if prompt is None:
    prompt = "What is shown in this image?"
query = [{"role": "user", "image": image, "content": prompt}]
</pre>